Sofa/framework/LinearAlgebra/src/sofa/linearalgebra/CompressedRowSparseMatrixMechanical.cpp

```diff
@@ -0,0 +1,381 @@
+}
+
+using namespace sofa::type;
+
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<float>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat1x1f>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat2x2f>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat3x3f>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat4x4f>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat<6, 6, float> >;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat<8, 8, float> >;
+
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<double>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat1x1d>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat2x2d>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat3x3d>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat4x4d>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat<6, 6, double> >;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat<8, 8, double> >;
+
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<float, CRSMechanicalStoreTouchedPolicy>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat1x1f, CRSMechanicalStoreTouchedPolicy>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat2x2f, CRSMechanicalStoreTouchedPolicy>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat3x3f, CRSMechanicalStoreTouchedPolicy>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat4x4f, CRSMechanicalStoreTouchedPolicy>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat<6, 6, float>, CRSMechanicalStoreTouchedPolicy >;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat<8, 8, float>, CRSMechanicalStoreTouchedPolicy >;
+
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<double, CRSMechanicalStoreTouchedPolicy>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat1x1d, CRSMechanicalStoreTouchedPolicy>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat2x2d, CRSMechanicalStoreTouchedPolicy>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat3x3d, CRSMechanicalStoreTouchedPolicy>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat4x4d, CRSMechanicalStoreTouchedPolicy>;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat<6, 6, double>, CRSMechanicalStoreTouchedPolicy >;
+template class SOFA_LINEARALGEBRA_API CompressedRowSparseMatrixMechanical<Mat<8, 8, double>, CRSMechanicalStoreTouchedPolicy >;
```

Comment:
I am not sure those should be instantiated.

Reply:
Even more, I would like to know if the CRSMechanicalStoreTouchedPolicy is (or would be) necessary itself.
The difference is that it enalbles StoreTouchFlags as a trait and it allows the usage of
```
    VecFlag  touchedBlock; ///< boolean vector, i-th value is true if block has been touched since last compression.
```
I dont know if it is necessary in our case ?

Reply:
For the moment, I would be in favor of removing the instantiations and the `CRSMechanicalStoreTouchedPolicy`. I don't know what it does and if it useful or not. To be investigated.